src/GovUk.Education.ExploreEducationStatistics.Content.Api/Services/MetaGuidanceService.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Content.Api.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Content.Api.ViewModels;
+using GovUk.Education.ExploreEducationStatistics.Data.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Publisher.Model.ViewModels;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using static GovUk.Education.ExploreEducationStatistics.Common.BlobContainerNames;
+
+namespace GovUk.Education.ExploreEducationStatistics.Content.Api.Services
+{
+    public class MetaGuidanceService : IMetaGuidanceService
+    {
+        private readonly IBlobStorageService _blobStorageService;
+        private readonly IMetaGuidanceSubjectService _metaGuidanceSubjectService;
+
+        public MetaGuidanceService(IBlobStorageService blobStorageService,
+            IMetaGuidanceSubjectService metaGuidanceSubjectService)
+        {
+            _blobStorageService = blobStorageService;
+            _metaGuidanceSubjectService = metaGuidanceSubjectService;
+        }
+
+        public async Task<Either<ActionResult, MetaGuidanceViewModel>> Get(string releasePath)
+        {
+            var text = await _blobStorageService.DownloadBlobText(
+                PublicContentContainerName,
+                releasePath
+            );
+
+            if (string.IsNullOrWhiteSpace(text))
+            {
+                throw new ArgumentException();
```

Comment:
`ArgumentException` by itself doesn't seem too helpful. Would suggest throwing something like `JsonException` with a descriptive message

Reply:
Cool, this highlighted a problem with `BlobStorageService` trying to open a stream on a blob with 0-byte content  even though `BlobClient.CanRead` is true and then receiving HTTP error 416 which resulted in a 500 error on our service leaking the implementation details. This is still an error but now handled with the JSONException as suggested.

Reply:
Ah nice, good catch!